torch/onnx/symbolic_opset13.py

```diff
@@ -771,9 +771,11 @@ def diagonal(g: jit_utils.GraphContext, self, offset, dim1, dim2):
     rank = symbolic_helper._get_tensor_rank(self)
     if rank is not None:
         axes = list(range(rank))
-        axes.remove(dim1)
-        axes.remove(dim2)
-        self = g.op("Transpose", self, perm_i=axes + [dim1, dim2])
+        dim1_real = dim1 if dim1 >= 0 else dim1 + rank
```

Comment:
Is it ok to have negative `dim1` and `dim2` on lines 759 and 762 above?

Would this work? This snippet re-computes `dim1` and `dim2` in the beginning if rank is available, otherwise, they remain as is. `dim1` and `dim2` would then be the same across the entire function

```python
@_onnx_symbolic("aten::diagonal")
@symbolic_helper.parse_args("v", "i", "i", "i")
@_beartype.beartype
def diagonal(g: jit_utils.GraphContext, self, offset, dim1, dim2):
    rank = symbolic_helper._get_tensor_rank(self)
    # Replace negative indexing when rank is known
    if rank is not None:
        dim1 = dim1 if dim1 >= 0 else dim1 + rank
        dim2 = dim2 if dim2 >= 0 else dim2 + rank

    dim1_size = opset9.size(
        g, self, dim=g.op("Constant", value_t=torch.LongTensor([dim1]))
    )
    dim2_size = opset9.size(
        g, self, dim=g.op("Constant", value_t=torch.LongTensor([dim2]))
    )
    # Create appropriate mask
    mask_shape = g.op("Concat", dim1_size, dim2_size, axis_i=0)
    mask = opset9.zeros(g, mask_shape, None, None, None)
    mask = g.op("EyeLike", mask, k_i=offset)
    # dim1 and dim2 appended as a dimension at the end of the shape
    rank = symbolic_helper._get_tensor_rank(self)
    if rank is not None:
        axes = list(range(rank))
        axes.remove(dim1)
        axes.remove(dim2)
        self = g.op("Transpose", self, perm_i=axes + [dim1, dim2])
    else:
        return symbolic_helper._unimplemented("diagonal", "unknown input rank")
```